src/services/services.ts

```diff
@@ -2365,6 +2369,40 @@ module ts {
                 return result;
             }
 
+            function isCompletionListBuilder(previousToken: Node): boolean {
+                if (previousToken) {
+                    var containingNodeKind = previousToken.parent.kind;
+                    switch (previousToken.kind) {
+                        case SyntaxKind.CommaToken:
+                            return containingNodeKind === SyntaxKind.CallExpression
+                                || containingNodeKind === SyntaxKind.Constructor;
+                        case SyntaxKind.OpenParenToken:
+                            return containingNodeKind === SyntaxKind.CallExpression
+                                || containingNodeKind === SyntaxKind.Constructor;
+                        case SyntaxKind.ModuleKeyword:
+                            return true;
+                        case SyntaxKind.DotToken:
+                            return containingNodeKind === SyntaxKind.ModuleDeclaration;
+                        case SyntaxKind.OpenBraceToken:
+                            return containingNodeKind === SyntaxKind.ClassDeclaration;
+                        case SyntaxKind.PublicKeyword:
+                        case SyntaxKind.PrivateKeyword:
+                        case SyntaxKind.ProtectedKeyword:
+                            return containingNodeKind === SyntaxKind.PropertyDeclaration;

```

Comment:
seems like this should be on anytime your containingNode is a class, interface, Enum or objectLiteral. e.g.:
```TypeScript
class A {
    a|
}
```